### PR TITLE
Users component, front and back end

### PR DESCRIPTION
**Pull request by ahillary**
* [x] no tests written
* [ ] front end view all and view single user page
* [ ] back end view all and view single user page
* [x] need seed data to test full functionality

### CODE REVIEW

File: client/components/users/UserList.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import {connect} from 'react-redux'
+import {fetchUsers, deleteUser} from '../../store/redux/robots/robotsReducer'
```
Comment (abstauff) [inline client/components/users/UserList.js]: Hey, you've got robots!

File: client/store/index.js [hunk 2]
```diff
@@ -1,13 +1,31 @@
 import {createStore, combineReducers, applyMiddleware} from 'redux'
 import {createLogger} from 'redux-logger'
+import axios from 'axios'
 import thunkMiddleware from 'redux-thunk'
 import {composeWithDevTools} from 'redux-devtools-extension'
 import user from './user'
```
Comment (abstauff) [inline client/store/index.js:7]: Import your reducer here

File: client/store/redux/users/usersReducer.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+export const setUsers = users => ({
+  type: SET_USERS,
+  users: users
+})
+
+/**
+ * THUNK CREATORS
+ */
+
+export const deleteUser = user => async (dispatch, getState, {axios}) => {
```
Comment (abstauff) [inline client/store/redux/users/usersReducer.js]: Potentially remove "getState, {axios}" from thunk creators

File: client/store/redux/users/usersReducer.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  switch (action.type) {
+    case DELETE_USER:
+      users = users.filter(
+        user => parseInt(user.id) !== parseInt(action.user.id)
+      )
+      return users
+    case SET_A_USER:
+      users = users.filter(
+        user => parseInt(user.id) !== parseInt(action.robot.id)
+      )
```
Comment (abstauff) [inline client/store/redux/users/usersReducer.js]: More robots! AH!

File: server/api/users.js [hunk 5]
```diff
@@ -1,17 +1,64 @@
...
-      attributes: ['id', 'email']
+    const user = await User.findAll({
+      // explicitly select only the id and email fields - even though users' passwords are encrypted, it is unnecessary to view more on the view all page
+      // Also, it won't help if we just send everything to anyone who asks!
+      attributes: ['id', 'firstName', 'lastName', 'email'],
     })
-    res.json(users)
-  } catch (err) {
-    next(err)
+    res.send(user)
```
Comment (abstauff) [inline server/api/users.js]: Change to res.json(user)

**Review by abstauff — CHANGES_REQUESTED** [review]
Great work, lookit you! Check out some comments to see what needs changing, and then we can make a merge!

File: client/components/users/UserList.js (hunk 1 above)
Comment (ahillary) [reply client/components/users/UserList.js]: fixed

File: client/store/redux/users/usersReducer.js (hunk 3 above)
Comment (ahillary) [reply client/store/redux/users/usersReducer.js:35]: imported

File: client/store/index.js (hunk 2 above)
Comment (ahillary) [reply client/store/index.js:7]: imported

File: client/store/redux/users/usersReducer.js (hunk 4 above)
Comment (ahillary) [reply client/store/redux/users/usersReducer.js]: fixed

File: server/api/users.js (hunk 5 above)
Comment (ahillary) [reply server/api/users.js]: updated

File: client/store/redux/users/usersReducer.js (hunk 3 above)
Comment (ahillary) [reply client/store/redux/users/usersReducer.js]: see new note in this file

File: server/api/index.js [hunk 6]
```diff
@@ -2,7 +2,7 @@ const router = require('express').Router()
 module.exports = router
 
 router.use('/users', require('./users'))
-router.use('/products', require('./products'))
+// router.use('/products', require('./products'))
```
Comment (abstauff) [inline server/api/index.js]: Uncomment please!

**Review by abstauff — CHANGES_REQUESTED** [review]
Uncomment the /products something

**Review by abstauff — APPROVED** [review]
YAY

**Pull request merged**